SPECS/cpprest/cpprest-2.10.14-Add-support-for-oauth2-using-client-credentials.patch

```diff
@@ -0,0 +1,131 @@
+From 708a5df2bb328705622c42f84b4167ea2c7c98c9 Mon Sep 17 00:00:00 2001
+From: Ivan Cherniukh <ivan.cherniukh@gmail.com>
+Date: Wed, 9 Sep 2020 09:09:04 -0700
+Subject: [PATCH] Add support for oauth2 using only client credentials
+
+---
+ .../cpprest/details/http_constants.dat        |  1 +
+ Release/include/cpprest/oauth2.h              | 15 ++++
+ .../functional/http/client/oauth2_tests.cpp   | 68 +++++++++++++++++++
+ 3 files changed, 84 insertions(+)
+
+diff --git a/Release/include/cpprest/details/http_constants.dat b/Release/include/cpprest/details/http_constants.dat
+index c3b1a53c..3deb24a1 100644
+--- a/Release/include/cpprest/details/http_constants.dat
++++ b/Release/include/cpprest/details/http_constants.dat
+@@ -190,6 +190,7 @@ DAT(expires_in,                     "expires_in")
+ DAT(grant_type,                     "grant_type")
+ DAT(redirect_uri,                   "redirect_uri")
+ DAT(refresh_token,                  "refresh_token")
++DAT(client_credentials,             "client_credentials")
+ DAT(response_type,                  "response_type")
+ DAT(scope,                          "scope")
+ DAT(state,                          "state")
+diff --git a/Release/include/cpprest/oauth2.h b/Release/include/cpprest/oauth2.h
+index 693ebbe3..68a7c7b9 100644
+--- a/Release/include/cpprest/oauth2.h
```

Comment:
Is this a public API change?

Reply:
This particular change to http_constants.dat is part of a private header (`include/cpprest/details`), so no. This adds a `const std::string` to a private helper class, as below (note the `details` namespace):

```
/// oAuth 2.0 library.
namespace oauth2
{
namespace details
{
class oauth2_handler;

// Constant strings for OAuth 2.0.
typedef utility::string_t oauth2_string;
class oauth2_strings
{
public:
#define _OAUTH2_STRINGS
#define DAT(a_, b_) _ASYNCRTIMP static const oauth2_string a_;
#include "cpprest/details/http_constants.dat"
#undef _OAUTH2_STRINGS
#undef DAT
};

} // namespace details
```

To add some more color, the added constant `client_credentials` is used in the impl of a newly added public function `token_from_client_credentials()` to add a uri parameter requiring a specific name for the oauth2 protocol. Since this is a new function that must be called explicitly, there should be no way for it to cause back-compat issues.